notebook/static/notebook/js/notebook.js

```diff
@@ -1647,9 +1647,8 @@ import {ShortcutEditor} from 'notebook/js/shortcuteditor';
     Notebook.prototype.render_cell_output = function (code_cell) {
         var cell_data = code_cell.toJSON();
         var cell_index = this.find_cell_index(code_cell);
-        this.delete_cell(cell_index);
-        var new_cell = this.insert_cell_at_index(cell_data.cell_type, cell_index);
-        new_cell.fromJSON(cell_data);
+        this.clear_output(cell_index);
```

Comment:
Ok, good call. So we have to either clear output, delete the cell, or something equivalent or `CodeCell.fromJSON` will add additional outputs to the cell. Do you have any other ideas?